pkg/scalers/azure_log_analytics_scaler.go

```diff
@@ -0,0 +1,490 @@
+	}
+
+	//Getting threshold
+	if val, ok := metadata["threshold"]; ok && val != "" {
+		threshold, err := strconv.ParseInt(val, 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse threshold: %s", err)
+		}
+		meta.threshold = threshold
+	} else if val, ok := metadata["thresholdFromEnv"]; ok && val != "" {
+		threshold, err := strconv.ParseInt(resolvedEnv[metadata["thresholdFromEnv"]], 10, 64)
+		if err != nil {
+			return nil, fmt.Errorf("can't parse threshold: %s", err)
+		}
+		meta.threshold = threshold
+	} else {
+		return nil, fmt.Errorf("threshold was not found in metadata. Check your ScaledObject configuration")
+	}
+
+	return &meta, nil
+}
+
+// IsActive determines if we need to scale from zero
+func (s *azureLogAnalyticsScaler) IsActive(ctx context.Context) (bool, error) {
+	err := s.updateCache()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+		return false, err
+	}
+
+	return s.cache.metricValue > 0, nil
+}
+
+func (s *azureLogAnalyticsScaler) GetMetricSpecForScaling() []v2beta2.MetricSpec {
+	err := s.updateCache()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+		return nil
+	}
+
+	externalMetric := &v2beta2.ExternalMetricSource{
+		Metric: v2beta2.MetricIdentifier{
+			Name: fmt.Sprintf("%s-%s", "azure-log-analytics", s.metadata.workspaceID),
+		},
+		Target: v2beta2.MetricTarget{
+			Type:         v2beta2.AverageValueMetricType,
+			AverageValue: resource.NewQuantity(s.cache.metricThreshold, resource.DecimalSI),
+		},
+	}
+	metricSpec := v2beta2.MetricSpec{External: externalMetric, Type: externalMetricType}
+	return []v2beta2.MetricSpec{metricSpec}
+}
+
+//GetMetrics returns value for a supported metric and an error if there is a problem getting the metric
+func (s *azureLogAnalyticsScaler) GetMetrics(ctx context.Context, metricName string, metricSelector labels.Selector) ([]external_metrics.ExternalMetricValue, error) {
+	receivedMetric, err := s.getMetricData()
+
+	if err != nil {
+		logAnalyticsLog.Error(err, "Failed to get metrics.")
+		return []external_metrics.ExternalMetricValue{}, err
+	}
+
+	metric := external_metrics.ExternalMetricValue{
+		MetricName: metricName,
+		Value:      *resource.NewQuantity(receivedMetric.value, resource.DecimalSI),
+		Timestamp:  metav1.Now(),
+	}
+
+	return append([]external_metrics.ExternalMetricValue{}, metric), nil
+}
+
+func (s *azureLogAnalyticsScaler) Close() error {
+	return nil
+}
+
+func (s *azureLogAnalyticsScaler) updateCache() error {
+	if s.cache.metricValue < 0 {
+		receivedMetric, err := s.getMetricData()
+
+		if err != nil {
+			logAnalyticsLog.Error(err, "Error processing Log Analytics query")
+			return err
+		}
+
+		s.cache.metricValue = receivedMetric.value
+
+		if receivedMetric.threshold > 0 {
+			s.cache.metricThreshold = receivedMetric.threshold
+		} else {
+			s.cache.metricThreshold = s.metadata.threshold
```

Comment:
what does `receivedMetric.threshold` represent here? All other scalers use the threshold value defined in the metadata for updating the HPA config. I'm trying to understand if supplying the threshold value from Azure Log Analytics makes sense or not here.

Reply:
There is a possibility to set threshold, based on query result. If query returns row with 2 cells, 2d cell will be interpreted as threshold. Query result has higher priority, than threshold, set in metadata. With this, threshold in metadata can be overwritten. I described this logic and limitations in a documentation for scaler.

Reply:
That makes sense. I just took a look at the docs PR. Thanks for the detailed explanation there.